### PR TITLE
[bitnami/kubernetes-event-exporter] add vpa support

### PR DESCRIPTION
**Pull request by t3mi**
### Description of the change
Adds support for Vertical Pod Autoscaler to automatically adjust cpu/memory requests/limits based on the actual usage without user intervention.

### Benefits
Improve cluster resource utilization.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by aoterolorenzo** [comment]
Hi @t3mi , thanks for the contribution!

Taking the opportunity, would you like to add also support to HPA in our `kubernetes-event-exporter`?

**Comment by t3mi** [comment]
@aoterolorenzo would prefer that to be done in a separate PR once https://github.com/resmoio/kubernetes-event-exporter/pull/34 merged.

File: bitnami/kubernetes-event-exporter/templates/vpa.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1/VerticalPodAutoscaler") .Values.autoscaling.vpa.enabled }}
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: {{ include "common.names.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
```
Comment (aoterolorenzo) [inline bitnami/kubernetes-event-exporter/templates/vpa.yaml:12]: As in other components, we include  k8's labels `component` and `part-of` in our vpa.yaml (in this case only the component one would proceed):

```
app.kubernetes.io/component: kubernetes-event-exporter
```

**Review by aoterolorenzo — CHANGES_REQUESTED** [review]

**Comment by aoterolorenzo** [comment]
> @aoterolorenzo would prefer that to be done in a separate PR once [resmoio/kubernetes-event-exporter#34](https://github.com/resmoio/kubernetes-event-exporter/pull/34) merged.

Sure! I just left one comment above in order to approve and merge it.

File: bitnami/kubernetes-event-exporter/templates/vpa.yaml (hunk 1 above)
Comment (t3mi) [reply bitnami/kubernetes-event-exporter/templates/vpa.yaml:12]: Added `part-of` label. I've used [`fluent-bit`](https://github.com/bitnami/charts/blob/main/bitnami/fluent-bit/templates/vpa.yaml#L12)  chart for the reference and it doesn't have it either.

**Review by aoterolorenzo — APPROVED** [review]

File: bitnami/kubernetes-event-exporter/templates/vpa.yaml (hunk 1 above)
Comment (aoterolorenzo) [reply bitnami/kubernetes-event-exporter/templates/vpa.yaml:12]: Thank you! My bad, I said it the other way around. Indeed, is `part-of`. Component would reference a subset of the application, like when we have an application with `migrator, exporter, what-ever-component...`. Thanks for noticing!

**Pull request merged**